common/components/utils/truncate.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+
+class Truncate {
+
+  static stringT(str: string, length: number) {
+    var dots = str.length > length ? "..." : "";
+    return str.substring(0, length) + dots;
+  }
+
+  static arrayT(arr: array, length: number) {
+    if (arr.length > length) {
+     let result = arr.slice(0, length-1)
+     result.push(arr.length - 3 + " more")
+     return result;
+    } else {
+      return arr;
+    }
+  }
+  static urlPrefixT(url: string) {
```

Comment:
nit: I wouldn't call this a truncation so much as a beautification.  url.js might be a more appropriate place to put it.